modules/end-user-portal/gpconnect-user-portal.test/Model/Apply/OrganisationModelTest.cs

```diff
@@ -0,0 +1,83 @@
+using GpConnect.NationalDataSharingPortal.EndUserPortal.Core.Config;
+using GpConnect.NationalDataSharingPortal.EndUserPortal.Core.Data.Interfaces;
+using GpConnect.NationalDataSharingPortal.EndUserPortal.Core.HttpClientServices.Interfaces;
+using GpConnect.NationalDataSharingPortal.EndUserPortal.Models;
+using GpConnect.NationalDataSharingPortal.EndUserPortal.Pages.Apply;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Options;
+using Moq;
+using System.Collections.Generic;
+using Xunit;
+using Microsoft.AspNetCore.Mvc.RazorPages;
+using System.Threading.Tasks;
+
+namespace GpConnect.NationalDataSharingPortal.EndUserPortal.Test.Pages.Apply;
+
+public class OrganisationModelTest
+{
+    private readonly Mock<IOrganisationLookupService> _mockOrganisationService;
+    private readonly Mock<ITempDataProviderService> _mockTempDataProviderService;
+    private readonly Mock<IOptions<ApplicationParameters>> _mockOptions;
+
+    public OrganisationModelTest()
+    {
+        _mockOrganisationService = new Mock<IOrganisationLookupService>();
+        _mockTempDataProviderService = new Mock<ITempDataProviderService>();
+        _mockOptions = new Mock<IOptions<ApplicationParameters>>();        
+    }
+
+    [Fact]
+    public async Task OnPostFind_IfValidEntry_RedirectsToNextPage()
+    {
+        var organisationModel = new OrganisationModel(_mockOptions.Object, _mockOrganisationService.Object, _mockTempDataProviderService.Object)
+        {
+            SiteOdsCode = "J82132"
+        };
+
+        var result = await organisationModel.OnPostFindOrganisationAsync();
+        Assert.IsType<RedirectToPageResult>(result);
+    }
+
+    [Fact]
+    public void OnPost_IfValidEntry_RedirectsToNextPage()
+    {
+        var organisationModel = new OrganisationModel(_mockOptions.Object, _mockOrganisationService.Object, _mockTempDataProviderService.Object)
+        {
+            SiteOdsCode = "J82132"
+        };
+
+        var result = organisationModel.OnPostNextAsync();
+        Assert.IsType<RedirectToPageResult>(result);
```

Comment:
Need to check the page in the RedirectToPageResult as this could be triggered by the './Timeout' redirect as well

Reply:
Sorted.